internal/project/project.go

```diff
@@ -205,6 +235,95 @@ func (p *Project) LanguageService() *ls.LanguageService {
 	return p.languageService
 }
 
+func (p *Project) getRootFileWatchGlobs() []string {
+	if p.kind == KindConfigured {
+		wildcardDirectories := p.parsedCommandLine.WildcardDirectories()
+		result := make([]string, 0, len(wildcardDirectories)+1)
+		result = append(result, p.configFileName)
+		for dir, recursive := range wildcardDirectories {
+			result = append(result, fmt.Sprintf("%s/%s", dir, core.IfElse(recursive, recursiveFileGlobPattern, fileGlobPattern)))
+		}
+		return result
+	}
+	return nil
+}
+
+func (p *Project) getModuleResolutionWatchGlobs() (failedLookups map[tspath.Path]string, affectingLocaions map[tspath.Path]string) {
+	failedLookups = make(map[tspath.Path]string)
+	affectingLocaions = make(map[tspath.Path]string)
+	for _, resolvedModulesInFile := range p.program.GetResolvedModules() {
+		for _, resolvedModule := range resolvedModulesInFile {
+			for _, failedLookupLocation := range resolvedModule.FailedLookupLocations {
+				path := p.toPath(failedLookupLocation)
+				if _, ok := failedLookups[path]; !ok {
+					failedLookups[path] = failedLookupLocation
+				}
+			}
+			for _, affectingLocation := range resolvedModule.AffectingLocations {
+				path := p.toPath(affectingLocation)
+				if _, ok := affectingLocaions[path]; !ok {
+					affectingLocaions[path] = affectingLocation
+				}
+			}
+		}
+	}
+	return failedLookups, affectingLocaions
+}
+
+func (p *Project) updateWatchers() {
+	client := p.host.Client()
+	if !p.host.IsWatchEnabled() || client == nil {
+		return
+	}
+
+	rootFileGlobs := p.getRootFileWatchGlobs()
+	failedLookupGlobs, affectingLocationGlobs := p.getModuleResolutionWatchGlobs()
+
+	if rootFileGlobs != nil {
+		if updated, err := p.rootFilesWatch.update(rootFileGlobs); err != nil {
+			p.log(fmt.Sprintf("Failed to update root file watch: %v", err))
+		} else if updated {
+			p.log("Root file watches updated:\n" + formatFileList(rootFileGlobs, "\t", hr))
+		}
+	}
+
+	if updated, err := p.failedLookupsWatch.update(failedLookupGlobs); err != nil {
+		p.log(fmt.Sprintf("Failed to update failed lookup watch: %v", err))
+	} else if updated {
+		p.log("Failed lookup watches updated:\n" + formatFileList(p.failedLookupsWatch.globs, "\t", hr))
+	}
+
+	if updated, err := p.affectingLocationsWatch.update(affectingLocationGlobs); err != nil {
+		p.log(fmt.Sprintf("Failed to update affecting location watch: %v", err))
+	} else if updated {
+		p.log("Affecting location watches updated:\n" + formatFileList(p.affectingLocationsWatch.globs, "\t", hr))
+	}
+}
```

Comment:
Need missing file watches .. On that note currently we add root files that are "present" but we need to be adding all root files and let program handle it so that it can issue "missing file" diagnostic and we can add watch as well